crates/protocol/src/traits.rs

```diff
@@ -0,0 +1,25 @@
+//! Traits for working with protocol types.
+
+use alloc::{boxed::Box, string::ToString};
+use async_trait::async_trait;
+use core::fmt::Display;
+use op_alloy_consensus::OpBlock;
+
+use crate::L2BlockInfo;
+
+/// Describes the functionality of a data source that fetches safe blocks.
+#[async_trait]
+pub trait BatchValidator {
```

Comment:
Slight preference for renaming this as `BatchValidator` is a stage in the pipeline now. Maybe `BatchValidationProvider`?

Reply:
I like this, will open a fresh stack... this pr is gonna be too heinous to review.